packages/amplify-cli/src/amplify-service-helper.ts

```diff
@@ -44,6 +43,16 @@ export const postPullCodegen = async (context: $TSContext): Promise<void> => {
     return;
   }
   if (await isDataStoreEnabled(path.join(pathManager.getBackendDirPath(), 'api', gqlApiName))) {
-    await context.amplify.invokePluginMethod(context, 'codegen', undefined, 'generateModels', [context]);
+    const { resourcesToBeCreated, resourcesToBeUpdated, resourcesToBeSynced, resourcesToBeDeleted } = await getResourceStatus(
+      AmplifyCategories.API,
+      undefined,
+      undefined,
+      undefined,
+    );
+    const apiHasChanges =
+      resourcesToBeCreated.length || resourcesToBeUpdated.length || resourcesToBeSynced.length || resourcesToBeDeleted.length;
```

Comment:
Do we need `resourcesToBeDeleted.length` ?
if api is getting deleted, do we have to call `generateModels` ?